README.md

```diff
@@ -1,30 +1,45 @@
-# Virtual list
+# &lt;virtual-list&gt;
 
-Implementation of a Virtual list API, influenced by the research done by the [infinite list study group](https://github.com/domenic/infinite-list-study-group).
+`<virtual-list>` renders only the visible portion of a list and optimizes the amount of DOM operations and reflows.
 
-## Local Setup
+## Properties
 
-Ensure you have installed the npm dependencies and serve from the project root
-```sh
-$ npm install
-$ polymer serve --npm
-```
-Then, navigate to the url: http://localhost:8081/components/virtual-list/
+- _newChild_
+  - type: `function(item: any, index: number) => Element`
+  - returns an `Element` for the given item and index.
+- _updateChild_
+  - type: `function(child: Element, item: any, index: number)`
+  - invoked after the `child` is appended to the DOM and requires to be updated with data.
+- _recycleChild_
+  - type: `function(child: Element, item: any, index: number)`
+  - invoked before the `child` is removed from the DOM. Set to control DOM removal/recycling behavior.
+- _items_
+  - type: `Array`
+  - the data model.
+- _layout_ 
+  - type: `string`
+  - layout type, can be set via attribute or property to "vertical" (default) or "horizontal".
 
-## Concepts and Domain
 
-An infinite list optimizes the rendering of DOM according to the visible area and available data.
+## Methods
 
-Some implementations append DOM incrementally, others recycle the DOM.
+- _requestReset()_
+  - Schedules a rerendering of the displayed items.
 
-## &lt;virtual-list&gt;
+## Events
 
-`<virtual-list>` can be configured through 3 properties:
-- `items (Array)`, the data model.
-- `template (Function|Object)`, generates the DOM for each data item. Can be set only once.
-- `direction (string)` (optional), layout direction, can be set via attribute or property to "vertical" (default) or "horizontal".
+- _rangechange_
+  - `bubbles: false, cancelable: false, composed: false`
+  - Fired when the list has rendered to a new range of items.
+  - _event.detail.first_
+    - type: `number`
+    - the index of the first item currently rendered.
+  - _event.detail.last_
+    - type: `number`
+    - the index of the last item currently rendered.

```

Comment:
This documentation is good. However, I would suggest (again it's fine to do as a follow-up) that instead of using CustomEvent, we subclass Event:

```js
class RangeChangeEvent {
  constructor(type, init) {
    super(type, init);
    this._first = init.first === undefined ? 0 : Math.floor(first);
    this._last = init.last === undefined ? 0 : Math.floor(first);
  }
  get first() {
    return this._first;
  }
  get last() {
}

Reply:
Nice idea, will include it in this PR